### DISCUSSION TITLE
Drag & drop on Android devices?

### DISCUSSION THREAD

**Discussion by janvorwerk** [adobe/react-spectrum#7023] [Q&A] [answered]
Hello,

I am experimenting drag & drop between two react-aria list-boxes, as presented here https://react-spectrum.adobe.com/react-aria/ListBox.html#drag-and-drop

I noticed the following weird behavior

It works like a charm on Chrome/macOS ✅

It works like a charm on iOS (Chrome & Safari) ✅

It does not work on Chrome/Android ❌... unless I specify a MIME type `text/plain`: using only `custom-app-type` prevents initiating the drag after the the long press.

Then, if I use both `text/plain` and `custom-app-type`, the drop target (in the same browser app) seems to only receive the `text/plain`.... and it squashes all items into a single newline separated string, as the Chrome DevTool screen capture shows:

<img width="607" alt="image" src="https://github.com/user-attachments/assets/11cd39de-b4fd-4717-ac62-7ed0357411d8">

Have you experienced the same behavior ? I suppose that since React Aria relies on the OS, there isn't much that can be done about, right?

## Version details

Pixel 5 device with Android 14 & Chrome 128

React Aria Components 1.3.3

**Comment by reidbarber** [comment, marked as answer]
See https://github.com/adobe/react-spectrum/issues/6365#issuecomment-2113565888

**Reply by janvorwerk** [reply]
Thank you! 🙏 I promise: I searched before asking 😇
